src/main/java/kr/hhplus/be/server/application/concert/ConcertService.java

```diff
@@ -41,4 +44,31 @@ public List<Seat> getAvailableSeatsByDate(LocalDate date) {
         return seatRepository.findAllByConcertSchedule_ScheduleIdAndStatus(schedule.getScheduleId(), SeatStatus.FREE);
     }
 
+    public List<ConcertSchedule> getSoldoutRanking(Integer year, Integer month, int topN) {
+        // 1. 연/월 없는 경우 이번 달
+        LocalDate now = LocalDate.now();
+        int y = (year != null) ? year : now.getYear();
+        int m = (month != null) ? month : now.getMonthValue();
+        String yearMonth = String.format("%04d%02d", y, m);
+
+        // 2. Redis에서 TOP N 랭킹 조회 -> schedule:{id} 중 id만 조회
+        List<Long> scheduleIds = concertSoldoutRankingRepository.getSoldoutRanking(yearMonth, topN)
+                .stream()
+                .map(member -> Long.parseLong(member.replace("schedule:", "")))
+                .toList();
+
+        // 3. 콘서트 일정 정보 조회
+        List<ConcertSchedule> schedules = scheduleRepository.findAllByScheduleIdIn(scheduleIds);
```

Comment:
아래와 같이scheduleIds가 비어있는 경우 불필요한 DB조회를 하지않도록 하는것이 좋을것 같습니다.

```suggestion
        List<Long> scheduleIds = concertSoldoutRankingRepository.getSoldoutRanking(yearMonth, topN)
                .stream()
                .map(member -> Long.parseLong(member.replace("schedule:", "")))
                .toList();

if (scheduleIds.isEmpty()) {
    return Collections.emptyList();
}

        // 3. 콘서트 일정 정보 조회
        List<ConcertSchedule> schedules = scheduleRepository.findAllByScheduleIdIn(scheduleIds);
```